### PR TITLE
Feat: added pagination to server

### PR DESCRIPTION
**Pull request by EugeneTeu**
Add pagination for server. Deals with #55

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -287,33 +287,41 @@ OR
...
+        "userId": "5f7d327766aa52759df235fe",
+        "answerIds": [],
+        "level": "primary",
+        "subject": "mathematics",
+        "upvotes": 0,
+        "downvotes": 0
+      },
+      // ...
+    ], 
+    "total": 15
```
Comment (EugeneTeu) [inline README.md:323]: Total here is to display to users "10 of 1000" at the bottom of the table

File: README.md [hunk 2]
```diff
@@ -287,33 +287,41 @@ OR
...
-      "answerIds": [],
-      "level": "primary",
-      "subject": "mathematics",
-      "upvotes": 0,
-      "downvotes": 0
-    },
-    // ...
-  ]
+  {
+    [
```
Comment (AaronCQL) [inline README.md:305]: The array is missing the `questions` key

File: server/src/controllers/questions.ts [hunk 3]
```diff
@@ -10,12 +10,27 @@ import {
...
   toValidObjectId,
 } from "../utils";
+import { GetQuestionRequestResponse } from "src/utils/types/GetQuestionRequestResponse";
+
+// TODO: add search/filter in the future
+async function getQuestions(
+  page: number,
+  pageSize: number
+): Promise<GetQuestionRequestResponse> {
+  if (!page || !pageSize || page < 0 || pageSize < 0) {
```
Comment (AaronCQL) [inline server/src/controllers/questions.ts:20]: No need to check `page < 0` or `pageSize < 0`, the `!page` and `!pageSize` already handles these cases.

File: server/src/controllers/questions.ts [hunk 4]
```diff
@@ -10,12 +10,27 @@ import {
...
+  }
 
-// TODO: add pagination/search/filter in the future
-async function getQuestions(): Promise<Question[]> {
-  const questions: Question[] = await getQuestionsCollection().find().toArray();
-
-  return questions;
+  const questions: Question[] = await getQuestionsCollection()
+    .find()
+    .skip(page > 0 ? (page - 1) * pageSize : 0)
```
Comment (AaronCQL) [inline server/src/controllers/questions.ts:29]: Likewise for this, no need for the ternary, `page` is guaranteed to be `>= 1`

**Review by AaronCQL — CHANGES_REQUESTED** [review]
Make sure tests pass :smile:

**Review by AaronCQL — APPROVED** [review]
LGTM

**Pull request merged**